src/pocketmine/level/Level.php

```diff
@@ -1862,11 +1862,15 @@ public function useItemOn(Vector3 $vector, Item &$item, int $face, Vector3 $clic
 
 			$ev->call();
 			if(!$ev->isCancelled()){
-				if(!$player->isSneaking() and $blockClicked->onActivate($item, $player)){
+				if(!$player->isSneaking()){
+					if($blockClicked->onActivate($item, $player)){
+						return true;
+					}elseif($item->onActivate($player, $blockReplace, $blockClicked, $face, $clickVector)){
+						return true;
+					}
+				}elseif($item->isNull() and $blockClicked->onActivate($item, $player)){
```

Comment:
can't this be simplified into the previous condition `(!$player->isSneaking() or $item->isNull())` , or am I missing some subtlety here?